### PR TITLE
Cannot modify diagram title in diagram editor #219

### PR DESCRIPTION
**Pull request by trrenty**

### CODE REVIEW

**Comment by mflorea** [comment]
Please upload a screenshot. Thanks.

**Comment by trrenty** [comment]
@mflorea 
image

**Comment by mflorea** [comment]
I'm not sure the icon before the "Title" label brings any value, on the contrary, it could be misleading / confusing. Plus it's not consistent (we don't use that icon elsewhere in XWiki before the title label). I think you should drop it. Otherwise it looks OK. Ideally, I would have liked the page title (heading) to be replaced by an "invisible" text input, similar to the in-place edit, in order to avoid the duplication (e.g. "DiagramTest" is displayed twice). But your solution works fine.

File: application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml [hunk 1]
```diff
@@ -75,6 +75,22 @@
   #set ($keyboardShortcutsEnabled = false)
   #set ($discard = $xwiki.ssx.use('Diagram.DiagramSheet'))
   #set ($discard = $xwiki.jsx.use('Diagram.DiagramEditSheet'))
+  ## Issue #219: Cannot modify diagram title in diagram editor
+  &lt;div class="row xform"&gt;
+    &lt;div class="col-xs-12 col-md-12"&gt;
```
Comment (mflorea) [inline application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml:80]: I think the ``col-md-`` makes sense only if it has a different value than ``col-xs-``. Basically, the larger screens use by default the same styles as the smaller screens.

File: application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml [hunk 2]
```diff
@@ -75,6 +75,22 @@
...
+      &lt;dl&gt;
+        &lt;dt&gt;
+          &lt;label for="xwikiDocTitleInput"&gt;
```
Comment (mflorea) [inline application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml:83]: I know you tried to follow the best practice by applying camel-case but I think it's better in this case to use the same id as in standard wiki edit mode. I know that id is awful but I think consistency is more important in this case.

File: application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml [hunk 3]
```diff
@@ -75,6 +75,22 @@
...
+            $services.localization.render('core.editors.content.titleField.label')
```
Comment (mflorea) [inline application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml:84]:
```suggestion
            $escapetool.xml($services.localization.render('core.editors.content.titleField.label'))
```

Always escape content that you don't control directly. In this case, the translation message can easily break the HTML.

File: application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml [hunk 4]
```diff
@@ -75,6 +75,22 @@
...
+          &lt;/label&gt;
+        &lt;/dt&gt;
+        &lt;dd&gt;
+          &lt;input id="xwikiDocTitleInput" type="text" name="title" value="$escapetool.xml($doc.title)"&gt;
```
Comment (mflorea) [inline application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml:88]: I know it doesn't make much sense to translate structured pages, such as diagram pages, but it is possible. Sure, all translations will share the same diagram object but they can have different titles, so it's best to edit the title corresponding to the current translation:

```suggestion
          &lt;input id="xwikiDocTitleInput" type="text" name="title" value="$escapetool.xml($tdoc.title)"&gt;
```

``tdoc`` stands for "translated" document.

**Review by mflorea — CHANGES_REQUESTED** [review]

**Pull request merged**